rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/rascal/model/PathConfigs.java

```diff
@@ -78,6 +79,41 @@ private PathConfig buildPathConfig(ISourceLocation projectRoot) {
         }
     }
 
+    private void recalcPathConfigIfChanged(ISourceLocation projectRoot, URIResolverRegistry reg, ISourceLocation targetFile)
+        throws IOException {
+        reg.watch(targetFile, false, changedManifest ->
+            currentPathConfigs.replace(projectRoot, actualBuild(projectRoot))
+        );
+    }
+
+    private void registerMavenWatches(URIResolverRegistry reg, ISourceLocation projectRoot) throws IOException {
+        var mainPom = URIUtil.getChildLocation(projectRoot, "pom.xml");
+        if (reg.exists(mainPom)) {
+            recalcPathConfigIfChanged(projectRoot, reg, mainPom);
+            if (hasParentSection(reg, mainPom)) {
+                var parentPom = URIUtil.getChildLocation(URIUtil.getParentLocation(projectRoot), "pom.xml");
+                if (!mainPom.equals(parentPom) && reg.exists(mainPom)) {
+                    recalcPathConfigIfChanged(projectRoot, reg, parentPom);
+                }
+            }
+        }
+    }
+
+    private static boolean hasParentSection(URIResolverRegistry reg, ISourceLocation mainPom) {
+        try (var pom = new BufferedReader(reg.getCharacterReader(mainPom))) {
+            String line;
+            while ((line = pom.readLine()) != null) {
+                if (line.contains("<parent>")) {
```

Comment:
this is a bit brittle, since people may write `< parent >` or `<parent\n>` or any other kind of whitespace that is allowed in XML.

Reply:
Okay, but then I should update the code to parse it as xml, and then search for the parent tag. In the end I wanted to do something that catches 99% of the cases, but if you feel this heuristic is too basic, I'll try to find some time to improve it.

Reply:
If we'd had the Maven API we could just query for this info. So let's leave it as-is and revisit the feature when we have a full blown Maven API bundled with the Rascal run-time.